src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -192,10 +200,11 @@ SnapTrimEvent::start()
         });
       });
     });
-  }, [this](std::exception_ptr eptr) -> snap_trim_ertr::future<seastar::stop_iteration> {
+  }, [this, ref=std::move(ref)]
+     (std::exception_ptr eptr) -> snap_trim_ertr::future<seastar::stop_iteration> {
     logger().debug("{}: interrupted {}", *this, eptr);
     return crimson::ct_error::eagain::make();
-  }, pg).finally([this] {
+  }, pg).finally([this, ref=std::move(ref)] {
```

Comment:
Isn't ref moved from 3 times in this function?  Seems like either they should be copying ref or the other two should capture *ref by reference rather than ref by move.

Reply:
Updated to copying